share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
Actually, I realized last night that putting everything into Boogie as I suggested above potentially has its own issues that would be nice to avoid. The issues I envision are related to manipulating pthreads structures outside of the API invocations. For example, can `pthread_mutex_t` structure be copied using `memcpy`? Seems crazy, but I guess it can be done, right? Things like this could potentially break a purely Boogie model since we would lose track of such relationships. I will meet with Monty this week to discuss this.